ShellPkg/Library/UefiShellLevel1CommandsLib/For.c

```diff
@@ -687,6 +690,10 @@ ShellCommandRunFor (
       }
 
       TempString = AllocateZeroPool (50*sizeof (CHAR16));
+      if (TempString == NULL) {
+        return (SHELL_OUT_OF_RESOURCES);

```

Comment:
`Info` might need to be freed (maybe other variables aswell)

Reply:
Agreed. Also freeing `ArgSet` here. Does not look like anything else needs to be freed.